core/src/main/scala/org/http4s/parser/Rfc3986Parser.scala

```diff
@@ -82,7 +82,9 @@ private[parser] trait Rfc3986Parser
 
   def Port = rule {
     ":" ~ (capture(oneOrMore(Digit)) ~> { s: String =>
-      (Some(s.toInt))
+      val int: Option[Int] = Either.catchOnly[NumberFormatException](s.toInt).toOption
```

Comment:
I was able to use your suggestion, https://github.com/http4s/http4s/pull/2664#discussion_r296258166, @SystemFw. Thanks for it!